pkg/interceptors/cel/cel.go

```diff
@@ -0,0 +1,183 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cel
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"reflect"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/checker/decls"
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
+	"github.com/google/cel-go/interpreter/functions"
+	"github.com/tektoncd/triggers/pkg/interceptors"
+	"github.com/tidwall/sjson"
+	"go.uber.org/zap"
+	exprpb "google.golang.org/genproto/googleapis/api/expr/v1alpha1"
+	"k8s.io/client-go/kubernetes"
+
+	triggersv1 "github.com/tektoncd/triggers/pkg/apis/triggers/v1alpha1"
+)
+
+// Interceptor implements a CEL based interceptor, that uses CEL expressions
+// against the incoming body and headers to match, if the expression returns
+// a true value, then the interception is "successful".
+type Interceptor struct {
+	KubeClientSet          kubernetes.Interface
+	Logger                 *zap.SugaredLogger
+	CEL                    *triggersv1.CELInterceptor
+	EventListenerNamespace string
+}
+
+// NewInterceptor creates a prepopulated Interceptor.
+func NewInterceptor(cel *triggersv1.CELInterceptor, k kubernetes.Interface, ns string, l *zap.SugaredLogger) interceptors.Interceptor {
+	return &Interceptor{
+		Logger:                 l,
+		CEL:                    cel,
+		KubeClientSet:          k,
+		EventListenerNamespace: ns,
+	}
+}
+
+// ExecuteTrigger is an implementation of the Interceptor interface.
+func (w *Interceptor) ExecuteTrigger(payload []byte, request *http.Request, _ *triggersv1.EventListenerTrigger, _ string) ([]byte, error) {
+	env, err := makeCelEnv()
+	if err != nil {
+		return nil, err
```

Comment:
Could we add some extra context when returning errors in this file? 
e.g. `fmt.Errof("error creating cel environment: %w, err)`

Reply:
Done